2021/w47_faster_software_through_register_based_calling.md

```diff
@@ -1,48 +1,48 @@
-# Faster software through register based calling
+# 基于寄存器调用的软件加速
 
 - 原文地址：https://menno.io/posts/golang-register-calling/
 - 原文作者：[Menno Finlay-Smits](https://github.com/mjs)
 - 本文永久链接：https://github.com/gocn/translator/blob/master/2021/w47_faster_software_through_register_based_calling.md
 - 译者：[cvley](https://github.com/cvley)
 - 校对：
 
-Posted on November 23, 2021
+2021 年 11 月 23 日发表
 
-The [release notes for Go 1.17](https://golang.org/doc/go1.17#compiler) mention an interesting change in the Go compiler: function arguments and return values will now be passed using registers instead of the stack. The [proposal document](https://go.googlesource.com/proposal/+/refs/changes/78/248178/1/design/40724-register-calling.md) for the feature mentions an expected 5-10% throughput improvement across a range of applications which is significant, especially for no effort on the developers part aside from recompiling with a newer version of the Go compiler. I was curious to see what this actually looks like and decided to take a deeper look. This will get more than a little nerdy so buckle up!
+ [Go 1.17 的发布说明](https://golang.org/doc/go1.17#compiler) 中提到一个 Go 编译器有趣的变更：现在将使用寄存器来代替栈，用于传递函数的参数和返回值。这个特性的 [提案文档](https://go.googlesource.com/proposal/+/refs/changes/78/248178/1/design/40724-register-calling.md) 提到，大量的应用预期可以提升 5-10% 的吞吐，尤其是对于开发者而言，除了使用新版的 Go 编译器重新编译外，没有其他的工作量。我好奇这个变更的原理，因此决定深入研究一番。接下来的旅程多少有些技术宅的气息，那么坐好我们开始吧！
```

Comment:
function arguments and return values will now be passed using registers instead of the stack
【个人想法】现在将使用寄存器的方式来传递方法函数和返回值，而不是堆来传递。
【原文】现在将使用寄存器来代替栈，用于传递函数的参数和返回值。

Reply:
The [proposal document] for the feature mentions an expected 5-10% throughput improvement across a range of applications which is significant，especially for no effort on the developers part aside from recompiling with a newer version of the Go compiler.
【个人想法】这个提案文章预期，在未来会有大量的应用在吞吐量性能方面会有5-10%的显著提升，特别的是只需要开发者重新编译一下新的应用版本。
【原文】这个特性的 [提案文档] 提到，大量的应用预期可以提升 5-10% 的吞吐，尤其是对于开发者而言，除了使用新版的 Go 编译器重新编译外，没有其他的工作量